### PR TITLE
Refactor stepper UI #253

### PR DESCRIPTION
**Pull request by belkassaby**

### CODE REVIEW

**Comment by belkassaby** [comment]
This PR is an improvement on the way the stepper is implemented. There is a current toggle button which allows toggling/changing the profile while in the stepper. The default profile is the rare disease but clicking on the toggle will show all possible steps. We can comment out that button if need be. this is deployed on phenopacketlab-dev.jax.org

**Comment by belkassaby** [comment]
Also, @ielis this is a question for you: here I just added the file to the stepper UI, where I'm re-using the existing File component that can be viewed when a phenopacket has been created (the accordion element before metadata) that has an "add file" button and then enables users to create Files. 
I was thinking of simplifying the code base by re-using the different components of a Phenopacket (with their corresponding tables) in the place of the almost similar steps (that have tables with less details columns). here is an example of what we currently have: for diseases:
Screenshot 2023-07-22 at 8 16 31 AM
 and in the stepper:

Screenshot 2023-07-22 at 8 16 56 AM

The changes I would do would make it look similar to the files where the same component is re-used

Screenshot 2023-07-22 at 8 17 49 AM

Screenshot 2023-07-22 at 8 18 36 AM

**Review by ielis — APPROVED** [review]
The reuse of UI is great, I hope this pattern will continue to work.

I tried to use the file UI and it worked OK. I only have few minor notes. There are some key-value pairs that get special treatment from right reason: `genomeAssembly`, `fileFormat`, and `description`. While the last can take free text, we should help the user more with choosing values for the first two. The user should be offered a dropdown with the following options:
- `genomeAssembly`: `{GRCh38, GRCh37}`
- `fileFormat`: `{VCF, BED, CRAM, BAM, SAM}`

However, at the same time, the dropdown should allow entering free text, e.g. if the genome build is not offered, etc. Something like an [editable dropdown](https://www.primefaces.org/primeng-v11-lts/#/dropdown).

Last, I am a bit worried concerning the profile switch (mendelian -> all -> mendelian -> etc.). I do not think this will scale well if we add more profiles. Moreover, a data loss can happen, e.g. if switching from *all* to *mendelian*. It can stay as a transient feature to simplify development but "hot profile switching" may need another thought..

**Comment by BethSundberg** [comment]
@belkassaby On the dev site, I can see the change in the rare disease/all steps button. This is working as expected if you start by using the "create" button. However, if you start with the "+ add phenopacket" button, it begins by showing the "show rare disease steps" button. The first time you click it there is no change to the stepper. This could be confusing. Can you make it start with the "show all available steps" button in this case. 
image

File: phenopacketlab-frontend/src/app/component/pheno-creator/pheno-creator.component.html [hunk 1]
```diff
@@ -1,7 +1,54 @@
...
   <div class="field col-12">
-    <h1 class="header-step"><b>Phenopacket Creator for Rare Disease</b></h1>
-    <p-steps [model]="items" [readonly]="false" ></p-steps>
+    <h1 class="header-step">
+      <div class="flex justify-content-between flex-wrap m-5">
+        <div class="flex align-items-center justify-content-center">
+          <b>Phenopacket Creator</b>
+        </div>
+        <div class="flex align-items-center justify-content-center">
+          <p-toggleButton
```
Comment (iimpulse) [inline phenopacketlab-frontend/src/app/component/pheno-creator/pheno-creator.component.html:9]: Can you make this button smaller it looks absolutely gigantic on the page.

File: phenopacketlab-frontend/src/app/component/pheno-creator/pheno-creator.component.html [hunk 2]
```diff
@@ -1,7 +1,54 @@
...
+            styleClass="p-button-outlined"
+            (onChange)="updateProfile($event.checked)"
+          ></p-toggleButton>
+        </div>
+      </div>
+      
+    </h1>
+    
+    <p-steps
+      [model]="items"
```
Comment (iimpulse) [inline phenopacketlab-frontend/src/app/component/pheno-creator/pheno-creator.component.html:24]: Can we name "items" to something more direct like steps

File: phenopacketlab-frontend/src/app/component/pheno-creator/pheno-creator.component.html [hunk 3]
```diff
@@ -1,7 +1,54 @@
...
+      [readonly]="false"
+      [(activeIndex)]="activeIndex"
+      (activeIndexChange)="onActiveIndexChange($event)"
+    ></p-steps>
   </div>
 </div>
 <router-outlet></router-outlet>
+<div class="flex justify-content-between flex-wrap m-5">
```
Comment (iimpulse) [inline phenopacketlab-frontend/src/app/component/pheno-creator/pheno-creator.component.html:32]: Can you make these buttons sticky to the bottom?

Comment (belkassaby) [reply phenopacketlab-frontend/src/app/component/pheno-creator/pheno-creator.component.html]: Just did it, let me know if you find that solution enough or if you have a more elegant solution for sticking these to the bottom of the step page
